### PR TITLE
Implement bozziplier

### PR DESCRIPTION
**Pull request by jchiam**
Implemented a bozz multiplier (bozziplier) where a sub-1 multiplier will be applied to all bozzcoin contributions should bozz not exercise for a certain number of days, determined by `bozziplierThreshold`.

Bozziplier is updated before every exercise entry if they're positive contributions. 

Bozziplier is reset to 1 every time bozz contributes.

Future: might want to set a minimum number of bozzcoins contributed by bozz within the threshold before the multiplier resets

### CODE REVIEW

**Comment by abeltay** [comment]
I believe a multiplier isn't a good reflection of effort because the multiplier could be high, but no one contributes, so the effect isn't felt or if the multiplier is low, but a lot of people contribute, resulting in a high penalty. I propose adding an increasing decay to the stack. For example, day 1 of non-activity will result in a decay of 10 bozzcoins, day 2 will result in 20, this could be upgraded to an exponential decay for more motivation.

**Comment by jchiam** [comment]
I think we can seek more opinions from the ppl in the slack channel. Personally I'm fine either way because as discussed over lunch yesterday, our sensing is that our exercise patterns are generally independent of bozzcoins, so whatever bozzcoin earning patterns we see will still be roughly what we get even with the multiplier on.

Using the multiplier gives the additional effect of multiple reminders to bozz as well since the pain of the multiplier is felt every time someone contributes :p That said, the logarithmic backoff of 0.5 might be a little too harsh.

File: scripts/bozzcoin.coffee [hunk 1]
```diff
@@ -79,9 +95,12 @@ module.exports = (robot) ->
         if repsDone > 13000
           res.send "Are you sure you can play longer than :pohneo:?"
           return
+    if repsDone > 0 && updateBozziplier(username)
+      res.send ":bozz: Bozziplier reset like a bozz"
+    bozziplier = robot.brain.get("bozziplier")
     newBozzcoinBalance = robot.brain.get("bozzcoinBalance") + convertToBozzcoin(repsDone, exerciseType)
     robot.brain.set("bozzcoinBalance", newBozzcoinBalance)
-    res.send "#{res.match[1]} #{exerciseType} done, *#{newBozzcoinBalance}* :bozzcoin: available!"
+    res.send "#{res.match[1]} #{exerciseType} done with bozzplier of #{bozziplier}, *#{newBozzcoinBalance}* :bozzcoin: available!"
```
Comment (fonglh) [inline scripts/bozzcoin.coffee]: Should be spelt `bozziplier` with the middle `i`.

Also, can add the amount of bozzcoin earned? Something like:
```
10 pullups done, earned 100 :bozzcoin: with bozziplier of 1.
4125 :bozzcoin: available.
```

Comment (jchiam) [reply scripts/bozzcoin.coffee]: Fixed and modified.

File: scripts/bozzcoin.coffee [hunk 2]
```diff
@@ -25,9 +28,22 @@ module.exports = (robot) ->
...
+  # returns true if bozziplier is reset
+  updateBozziplier = (username) ->
+    if username is "rurouni"
+      robot.brain.set("bozzLastExercised", new Date())
+      return true
+    else
+      bozzLastExercised = robot.brain.get("bozzLastExercised")
+      if !bozzLastExercised || new Date().getTime() - new Date(bozzLastExercised).getTime() > bozziplierThreshold * 86400000 # threshold times milliseconds in a day
+        bozziplier = robot.brain.get("bozziplier")
+        newBozziplier = bozziplier * 0.5
```
Comment (fonglh) [inline scripts/bozzcoin.coffee]: 0.5 is really harsh, that decays to 0.007 in 1 week.

Perhaps use 0.9? That's 0.478 after 1 week.

**Pull request merged**